File: Treachery-iOS/Treachery-iOS/Models/ExecutableAbility.swift
//
//  ExecutableAbility.swift
//  Treachery-iOS
//

import Foundation

enum ExecutableAbility: String, CaseIterable {
    case metamorph = "traitor_07"
    case puppetMaster = "traitor_09"
    case wearerOfMasks = "traitor_13"

    init?(cardId: String) {
        self.init(rawValue: cardId)
    }

    var displayName: String {
        switch self {
        case .metamorph: return "The Metamorph"
        case .puppetMaster: return "The Puppet Master"
        case .wearerOfMasks: return "The Wearer of Masks"
        }
    }
}

File: TreacheryAndroid/app/src/main/java/com/solomon/treachery/ui/game/ExecutableAbility.kt
package com.solomon.treachery.ui.game

// The three traitor identities whose unveil triggers a callable-backed
// follow-up (resolveMetamorph / resolvePuppetMaster / resolveWearerOfMasks).
// Card ids mirror ExecutableAbility.swift (iOS) and AbilityResolver.tsx (web).
enum class ExecutableAbility(val cardId: String) {
    METAMORPH("traitor_07"),
    PUPPET_MASTER("traitor_09"),
    WEARER_OF_MASKS("traitor_13");

    val displayName: String
        get() = when (this) {
            METAMORPH -> "The Metamorph"
            PUPPET_MASTER -> "The Puppet Master"
            WEARER_OF_MASKS -> "The Wearer of Masks"
        }

    companion object {
        fun fromCardId(cardId: String?): ExecutableAbility? =
            cardId?.let { id -> entries.find { it.cardId == id } }
    }
}
